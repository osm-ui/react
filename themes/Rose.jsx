
import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import _ from 'lodash';
import colors from '../colors';
import { buildDarkThemeConfig } from '../helpers/themes';
import {
    config as defaultThemeConfig,
    StyledDiv,
} from './Default';


export const config = _.merge(
    {},
    buildDarkThemeConfig(defaultThemeConfig, colors, 'rose'),
    {},
);


const RoseTheme = ({ children, ...rest }) => (
    <ThemeProvider theme={config}>
        <StyledDiv {...rest}>
            {children}
        </StyledDiv>
    </ThemeProvider>
);


RoseTheme.propTypes = {
    children: PropTypes.node.isRequired,
};

RoseTheme.displayName = 'RoseTheme';

export default RoseTheme;
