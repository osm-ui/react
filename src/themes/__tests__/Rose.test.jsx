import React from 'react';
import {
  snapshotWithElementChildren,
  snapshotWithTextChildren
} from 'helpers/tests';
import Rose from 'themes/Rose';

describe('When using snapshots', () => {
  it('Should render with an element children', () =>
    snapshotWithElementChildren(Rose));

  it('Should render with a text children', () =>
    snapshotWithTextChildren(Rose));
});
