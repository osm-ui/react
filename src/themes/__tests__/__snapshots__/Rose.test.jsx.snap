// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When using snapshots Should render with a text children 1`] = `
<styled.span>
  <RoseTheme
    key=".0"
  >
    This is a text
  </RoseTheme>
</styled.span>
`;

exports[`When using snapshots Should render with an element children 1`] = `
<styled.span>
  <RoseTheme
    key=".0"
  >
    <div />
  </RoseTheme>
</styled.span>
`;
